worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -194,10 +194,14 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
 }
 
 function onAfterProcessDocument(e) {
-    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately()) {
-        traverseDocumentForFailures(e.document);
-    } else {
-        processFailures(e.document);
+    if (fieldExists(e.document, "CAF_WORKFLOW_ACTION") &&
+            e.document.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0) !== null &&
+            !getTerminateOnFailure(e.document.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0))) {
```

Comment:
- The `CAF_WORKFLOW_ACTION` is only added to the root document as far as I know, so I think this should be `e.rootDocument` instead of `e.document`.  Actually that is probably also a fault in the `processFailures()` function as well.
- `.getStringValues().get(0) !== null`: Given that you've already established that the field has values, what are you trying to do here?
- Perhaps I'm missing something but I don't see how this solves the problem for the final worker.  (I can only see how how it works for a non-final worker that has `terminateOnFailure` turned on.)

Reply:
The main idea is,

- if this is not a terminateOnFailure worker -> process the failures as usual
- if this is a terminateOnFailure worker -> do not process them, and let the worker do its default action (put the message in the error queue)

Reply:
As discussed, the behavior when the flag is not set should depend on whether or not the action is the final action.